lib/buildkite/test_collector/ci.rb

```diff
@@ -69,17 +71,19 @@ def github_actions
       "branch" => ENV["GITHUB_REF_NAME"],
       "commit_sha" => ENV["GITHUB_SHA"],
       "number" => ENV["GITHUB_RUN_NUMBER"],
+      "build_id" => "#{ENV["GITHUB_RUN_ID"]}-#{ENV["GITHUB_RUN_ATTEMPT"]}"
```

Comment:
I think we might also need `GITHUB_ACTION` added to this value.

From the [GitHub docs](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables):
> A unique number for each workflow run **within** a repository.
That sounds like it's not unique outside the repo. There's nothing preventing someone from using TA with multiple repos so it might not be unique enough for our purposes.

What do you think?

Reply:
HMM GITHUB_ACTION doesn't sound unique either though:

> The name of the action currently running, or the id of a step. For example, for an action, __repo-owner_name-of-action-repo...
> GitHub removes special characters, and uses the name __run when the current step runs a script without an id...
> If you use the same script or action more than once in the same job, the name will include a suffix that consists of the sequence number preceded by an underscore [e.g] `__run_2`

I reckon using the GITHUB_REPOSITORY_ID would be the safest way to guarantee uniqueness given run_ids are scoped to them.